### PR TITLE
feat: add tls support in git-ops and git account (Global Configuration)

### PR DESCRIPTION
**Pull request by AbhishekA1509**
# Description
This PR includes changes to add TLS Connection support for GitOps and Git account forms in Global Configuration.
Screenshot 2024-06-13 at 5 03 56 PM

Fixes https://github.com/devtron-labs/devtron/issues/4659

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/common/TLSConnectionForm/types.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+export interface TLSConnectionDTO {
+    enableTLSVerification: boolean
+    tlsConfig: TLSConfigDTO
+    /**
+     * Used in case when we have already saved data and we have'nt filled the fields
+     */
+    isCADataPresent: boolean
+    isTLSCertDataPresent: boolean
+    isTLSKeyDataPresent: boolean
```
Comment (eshankvaish) [inline src/components/common/TLSConnectionForm/types.ts:33]: suggestion: these can be typed better

File: src/components/common/TLSConnectionForm/types.ts [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    updateAction:
+        | TLSConnectionFormActionType.UPDATE_CA_DATA
+        | TLSConnectionFormActionType.UPDATE_CERT_DATA
+        | TLSConnectionFormActionType.UPDATE_KEY_DATA
+
+    showClearButton?: boolean
+    clearAction:
+        | TLSConnectionFormActionType.CLEAR_CA_DATA
+        | TLSConnectionFormActionType.CLEAR_CERT_DATA
+        | TLSConnectionFormActionType.CLEAR_KEY_DATA
```
Comment (eshankvaish) [inline src/components/common/TLSConnectionForm/types.ts:80]: The typing for update and clear can be improved to have more coupled mapping

File: src/components/common/index.ts [hunk 3]
```diff
@@ -43,3 +43,4 @@ export * from './helpers/compareVersion'
 export * from './hooks/FileReader'
 export { default as FloatingVariablesSuggestions } from './FloatingVariablesSuggestions/FloatingVariablesSuggestions'
 export { default as HiddenInput } from './HiddenInput/HiddenInput'
+export * from './TLSConnectionForm'
```
Comment (eshankvaish) [inline src/components/common/index.ts:46]: let's remove this as discussed

**Review by eshankvaish — APPROVED** [review]

File: src/components/common/index.ts (hunk 3 above)
Comment (AbhishekA1509) [reply src/components/common/index.ts:46]: If going with this approach should delete whole file and have it as a refactoring task.

File: src/components/common/TLSConnectionForm/types.ts (hunk 1 above)
Comment (AbhishekA1509) [reply src/components/common/TLSConnectionForm/types.ts:33]: Have not hard coded the values for these anywhere, and it is being extended for input state so may not be a good idea.

**Review by vivek-devtron — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**